src/Persistence/MassTransit.Azure.Table/Contexts/AzureTableDatabaseContext.cs

```diff
@@ -0,0 +1,101 @@
+namespace MassTransit.Azure.Table
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Cosmos.Table;
+
+
+    public class AzureTableDatabaseContext<TSaga> :
+        DatabaseContext<TSaga>
+        where TSaga : class, IVersionedSaga
+    {
+        readonly string _partitionKey;
```

Comment:
Everything has the same partition key for a given saga type? Is that bad?

Reply:
My thinking was that this should be a small enough partition for the usecases which may end up using Table Storage (Fairly small load). But granted - if there ends up being over 10,000 instances of that Saga Type at once. Performance could degrade  from then on. But given the load is using PartitionKey & RowKey - it should be pretty snappy anyway!

Reply:
it should handle really well. I had experience using table storage with a lot records per same partition key. also it gives you ability to easily find saga by id. and get all by type

Reply:
I’m not fully sure it makes sense, better to split them trough tables, but azure table storage handles huge amount of data really well. we could disable using saga type per table, instead use all saga per single table but gives ability to generate partition key for saga type (same we did for mongodb with `ICollectionNameFormatter` for collection). Should solve the case when you rename your saga class name (I did it often) :D

Reply:
I considered splitting tables but it partition key because less meaningful in those settings as there would only ever be 1 partition in each table.

Ah! I hadn't considered the Saga rename issue!  I've tried to be deliberately pretty minimal with the first implementation (Only optimistic too) and keen to keep adding additional stuff where needs - that'd be first on the list!

Reply:
I've changed a bunch of things and will merge this shortly.